### PR TITLE
Update Android SDK to latest version

### PR DESCRIPTION
**Pull request by franrios**

### CODE REVIEW

File: android/src/main/java/com/robinpowered/react/Intercom/IntercomModule.java [hunk 1]
```diff
@@ -155,12 +155,6 @@ public void setUserHash(String userHash, Callback callback) {
     }
 
     @ReactMethod
-    public void setHMAC(String hmac, String data, Callback callback) {
-        Intercom.client().setSecureMode(hmac, data);
```
Comment (franrios) [inline android/src/main/java/com/robinpowered/react/Intercom/IntercomModule.java:159]: This method is not supported anymore

**Comment by franrios** [comment]
Intercom Android SDK 3.+ version was not working for us.

File: android/build.gradle [hunk 2]
```diff
@@ -34,5 +34,6 @@ repositories {
 
 dependencies {
     compile 'com.facebook.react:react-native:+'
-    compile 'io.intercom.android:intercom-sdk-base:3.+'
+    compile 'io.intercom.android:intercom-sdk-base:4.1.4'
```
Comment (jtreanor) [inline android/build.gradle]: Would using `4.+` be a good idea here?

Comment (franrios) [reply android/build.gradle]: I used that fixed version as Intercom support suggested. Anyway, we shouldn't have any problem using `4.+` so go for it 👍

**Comment by browniefed** [comment]
@jtreanor does this look good to go?

**Comment by jtreanor** [comment]
@browniefed Yep, it all looks good 👍

**Comment by browniefed** [comment]
Published as major v9

**Comment by slorber** [comment]
Hi,

I just started using this lib on a freshly ejected Expo project and wasn't able to sync Gradle and build the app because of this error:

```
Error:Execution failed for task ':app:processDebugManifest'.
> Manifest merger failed : Attribute meta-data#android.support.VERSION@value value=(26.1.0) from [com.android.support:design:26.1.0] AndroidManifest.xml:28:13-35
  	is also present at [com.android.support:exifinterface:26.0.1] AndroidManifest.xml:25:13-35 value=(26.0.1).
  	Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to override.
```

This does not look like an unresolvable problem, but as I don't know much about Android, maybe just the install instructions can be updated so that it works fine out of the box on Expo ejected projects?

Please note that I also tried with 8.1.0 and it works fine, so it's really a 9.0 related issue

**Comment by slorber** [comment]
Hi,

For those passing by, just be aware that fixing all android support library to the same version does prevent runtime crashes + solves the manifest merging problem I described above.

Check this for more details: https://forums.expo.io/t/detached-expo-on-android-sdk-build-tools-and-support-library-versions/5504

**Pull request merged**